components/SwapWidget.tsx

```diff
@@ -0,0 +1,318 @@
+  }, [platforms])
+
+  const onSwap = () => {
+    if (network && platforms[network]) {
+      const networkId = network === 'xdai' ? 100 : 1
+      const contractAddress = platforms[network].contractAddress
+
+      let sellToken, buyToken
+      if (activeTab === 'Buy') {
+        sellToken = networkId === 100 ? 'WXDAI' : 'WETH'
+        buyToken = contractAddress
+
+        if (contractAddress === WXDAI) {
+          sellToken = 'XDAI'
+        }
+
+        if (WETH.includes(contractAddress)) {
+          sellToken = networkId === 100 ? 'WXDAI' : 'ETH'
+        }
+      } else {
+        sellToken = contractAddress
+        buyToken = networkId === 100 ? 'WXDAI' : 'WETH'
+
+        if (contractAddress === WXDAI) {
+          buyToken = 'XDAI'
+        }
+
+        if (WETH.includes(contractAddress)) {
+          buyToken = networkId === 100 ? 'WXDAI' : 'ETH'
+        }
+      }
+
+      const url = `https://swap.cow.fi/#/${networkId}/swap/${sellToken}/${buyToken}?${activeTab.toLowerCase()}Amount=${amount}`
+      return url
+    } else {
+      return '#'
+    }
+  }
+
+  return (
+    <Wrapper>
+      <TabContainer>
+        <Tab onClick={() => setActiveTab('Buy')} active={activeTab === 'Buy'}>
+          Buy
+        </Tab>
+        <Tab onClick={() => setActiveTab('Sell')} active={activeTab === 'Sell'}>
+          Sell
+        </Tab>
+      </TabContainer>
+
+      <DropdownContainer>
+        <DropdownHeader onClick={() => setIsOpen(!isOpen)}>
+          <img
+            src={`/images/${network === Networks.ETHEREUM ? 'ethereum' : 'gnosis-chain'}.svg`}
+            alt={NETWORK_MAP[network]}
+          />
+          <b>{NETWORK_MAP[network]}</b>
+        </DropdownHeader>
+        {isOpen && (
+          <DropdownBody>
+            {platforms.ethereum.contractAddress && (
+              <DropdownOption onClick={() => handleSelect('ethereum')}>
+                <img src="/images/ethereum.svg" alt="Ethereum" />
+                Ethereum
+              </DropdownOption>
+            )}
+            {platforms.xdai.contractAddress && (
+              <DropdownOption onClick={() => handleSelect('xdai')}>
+                <img src="/images/gnosis-chain.svg" alt="Gnosis Chain" />
+                Gnosis Chain
+              </DropdownOption>
+            )}
+          </DropdownBody>
+        )}
+      </DropdownContainer>
+
+      <InputLabel>
+        {activeTab === 'Buy' ? 'Receive' : 'Send'}
+
+        <div>
+          <TokenLabel>
+            <img src={tokenImage} alt={tokenSymbol} />
+            <span>{tokenSymbol}</span>
+          </TokenLabel>
+          <Input type="number" onChange={(e) => setAmount(parseFloat(e.target.value))} placeholder="0" />
+        </div>
+      </InputLabel>
+
+      <Button
+        href={onSwap()}
+        target="_blank"
+        rel="noreferrer"
+        label={`Swap ${tokenSymbol}`}
+        fontSize={1.6}
+        minHeight={4.2}
+      />
+    </Wrapper>
+  )
+}
```

Comment:
In general this widget could use some cleaning / small refactor, before w take it to production imo